agenta-backend/agenta_backend/routers/app_router.py

```diff
@@ -284,6 +286,51 @@ async def create_app(
         raise HTTPException(status_code=500, detail=str(e))
 
 
+@router.put("/{app_id}/", response_model=UpdateAppOutput, operation_id="update_app")
+async def update_app(
+    app_id: str,
+    payload: UpdateApp,
+    request: Request,
+) -> UpdateAppOutput:
+    """
+    Update an app for a user or organization.
+
+    Args:
+        app_id (str): The ID of the app.
+        payload (UpdateApp): The payload containing the app name.
+        stoken_session (SessionContainer): The session container containing the user's session token.
+
+    Returns:
+        UpdateAppOuput: The output containing the newly created app's ID and name.
+
+    Raises:
+        HTTPException: If there is an error creating the app or the user does not have permission to access the app.
+    """
+
+    try:
+        app = await db_manager.fetch_app_by_id(app_id)
+        if isCloudEE():
+            has_permission = await check_action_access(
+                user_uid=request.state.user_id,
+                object=app,
+                permission=Permission.UPDATE_APPLICATION,
+            )
+            logger.debug(f"User has Permission to update app: {has_permission}")
+            if not has_permission:
+                error_msg = f"You do not have access to perform this action. Please contact your organization admin."
+                return JSONResponse(
+                    {"detail": error_msg},
+                    status_code=403,
+                )
+        await db_manager.update_app(
```

Comment:
Updating is always a tricky situation.

-- Who is authorized to update which part of a resourc/entity, and how ?

Here, we are indiscriminately allowing changes to any app attributes. Shouldn't we, at least, have a list of allowed update-able attributes ?

Reply:
The only attribute that can be updated is the application name; that’s the only attribute that can be updated in the `app_db` table.

Reply:
I'm not sure how _blocking_ this comment is. 
Maybe it isn't and we just open another ticket or discussion to talk about this more.
@mmabrouk @aybruhm -- up to you.